verifier/src/lib.rs

```diff
@@ -58,11 +95,66 @@ pub struct VerificationKey {
     pub vk_ic: Vec<[u8; G1_LEN]>,
 }
 
+/// The public inputs for a Groth16 proof.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct PublicInputs<const N: usize> {
     pub inputs: [[u8; SCALAR_LEN]; N],
 }
 
+/// The necessary information for a solana program to verify an SP1 Groth16 proof.
+///
+/// TODO: Add helper methods for extracting the program vkey and public inputs from `public_inputs`.
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct SP1ProofFixture {
+    pub proof: Vec<u8>,
+    pub public_inputs: Vec<u8>,
+}
+
+impl SP1ProofFixture {
+    pub fn load(path: impl AsRef<Path>) -> Result<Self, Error> {
+        let path = path.as_ref();
+        let file = File::open(path).map_err(|_| Error::IoError)?;
+        let mut reader = BufReader::new(file);
+        let fixture = borsh::from_reader(&mut reader).map_err(|_| Error::BorshDeserializeError)?;
+        Ok(fixture)
+    }
+
+    pub fn save(&self, path: impl AsRef<Path>) -> Result<(), Error> {
+        let path = path.as_ref();
+        let file = File::create(path).map_err(|_| Error::IoError)?;
+        let mut writer = BufWriter::new(file);
+        BorshSerialize::serialize(&self, &mut writer).map_err(|_| Error::BorshSerializeError)?;
+        Ok(())
+    }
+}
+
+#[cfg(feature = "sp1-serialize")]
```

Comment:
In order to import this crate to the solana runtime, we can't depend on sp1_sdk because it's too heavy. I made this flag so that users can still easily convert from native Sp1 groth16 serialization to a format better suiting for solana